diff --git a/plugins/core/IO/qE57IO/extern/libE57Format/include/E57Format.h b/plugins/core/IO/qE57IO/extern/libE57Format/include/E57Format.h
index 89032b1..3f9459f 100644
--- a/plugins/core/IO/qE57IO/extern/libE57Format/include/E57Format.h
+++ b/plugins/core/IO/qE57IO/extern/libE57Format/include/E57Format.h
@@ -32,6 +32,7 @@
 //! @file  E57Format.h header file for the E57 API
 
 #include <cfloat>
+#include <cstdint>
 #include <memory>
 #include <vector>
 
