:100644 100644 0a8a72b32 000000000 M	cmake/pcl_find_gl.cmake

diff --git a/cmake/pcl_find_gl.cmake b/cmake/pcl_find_gl.cmake
index 0a8a72b32..871fcca7d 100644
--- a/cmake/pcl_find_gl.cmake
+++ b/cmake/pcl_find_gl.cmake
@@ -7,7 +7,7 @@ if(POLICY CMP0072)
   cmake_policy(SET CMP0072 NEW)
 endif()
 
-find_package(OpenGL QUIET REQUIRED)
+find_package(OpenGL QUIET)
 
 if(APPLE AND OPENGL_FOUND)
   if("${OPENGL_INCLUDE_DIR}" MATCHES "\\.framework")
