diff --git a/plugins/core/Standard/q3DMASC/ContextBasedFeature.cpp b/plugins/core/Standard/q3DMASC/ContextBasedFeature.cpp
index ef9b066..78cc700 100644
--- a/plugins/core/Standard/q3DMASC/ContextBasedFeature.cpp
+++ b/plugins/core/Standard/q3DMASC/ContextBasedFeature.cpp
@@ -26,9 +26,6 @@
 //Qt
 #include <QMutex>
 
-#if defined(_OPENMP)
-#include <omp.h>
-#endif
 
 using namespace masc;
 
@@ -189,12 +186,6 @@ bool ContextBasedFeature::prepare(	const CorePoints& corePoints,
 			double meanNeighborhoodSize = 0;
 			int tenth = pointCount / 10;
 			bool cancelled = false;
-#ifndef _DEBUG
-#if defined(_OPENMP)
-			omp_set_num_threads(std::max(1, omp_get_max_threads() - 2));
-#pragma omp parallel for
-#endif
-#endif
 			for (int i = 0; i < static_cast<int>(pointCount); ++i)
 			{
 				const CCVector3* P = corePoints.cloud->getPoint(i);
diff --git a/plugins/core/Standard/q3DMASC/PointFeature.cpp b/plugins/core/Standard/q3DMASC/PointFeature.cpp
index 5e2d56a..5dd19b5 100644
--- a/plugins/core/Standard/q3DMASC/PointFeature.cpp
+++ b/plugins/core/Standard/q3DMASC/PointFeature.cpp
@@ -20,9 +20,6 @@
 //Local
 #include "q3DMASCTools.h"
 
-#if defined(_OPENMP)
-#include <omp.h>
-#endif
 
 //qPDALIO
 #ifdef PLUGIN_IO_QPDAL
@@ -422,12 +419,6 @@ static bool ComputeMathOpWithNearestNeighbor(	const CorePoints& corePoints,
 	double meanNeighborhoodSize = 0;
 	int tenth = pointCount / 10;
 	error.clear();
-#ifndef _DEBUG
-#if defined(_OPENMP)
-	omp_set_num_threads(std::max(1, omp_get_max_threads() - 2));
-#pragma omp parallel for
-#endif
-#endif
 	for (int i = 0; i < static_cast<int>(pointCount); ++i)
 	{
 		const CCVector3* P = corePoints.cloud->getPoint(i);
diff --git a/plugins/core/Standard/q3DMASC/q3DMASCClassifier.cpp b/plugins/core/Standard/q3DMASC/q3DMASCClassifier.cpp
index 7a7464a..3694d57 100644
--- a/plugins/core/Standard/q3DMASC/q3DMASCClassifier.cpp
+++ b/plugins/core/Standard/q3DMASC/q3DMASCClassifier.cpp
@@ -42,8 +42,8 @@
 #include "qTrain3DMASCDialog.h"
 #include "confusionmatrix.h"
 
-#if defined(_OPENMP)
-#include <omp.h>
+#if defined(CC_LINUX)
+#include <unistd.h>
 #endif
 
 using namespace masc;
@@ -202,12 +202,6 @@ bool Classifier::classify(	const Feature::Source::Set& featureSources,
 
 	bool success = true;
 	int numberOfTrees = static_cast<int>(m_rtrees->getRoots().size());
-#ifndef _DEBUG
-#if defined(_OPENMP)
-	omp_set_num_threads(std::max(1, omp_get_max_threads() - 2));
-#pragma omp parallel for
-#endif
-#endif
 	for (int i = 0; i < static_cast<int>(cloud->size()); ++i)
 	{
 		//allocate the data matrix
@@ -647,6 +641,7 @@ bool Classifier::train(	const ccPointCloud* cloud,
 #if defined(CC_WINDOWS)
 		::Sleep(500);
 #else
+
 		usleep(500 * 1000);
 #endif
 		if (pDlg)
diff --git a/plugins/core/Standard/q3DMASC/q3DMASCDisclaimerDialog.h b/plugins/core/Standard/q3DMASC/q3DMASCDisclaimerDialog.h
index 873fa15..945d5ba 100644
--- a/plugins/core/Standard/q3DMASC/q3DMASCDisclaimerDialog.h
+++ b/plugins/core/Standard/q3DMASC/q3DMASCDisclaimerDialog.h
@@ -25,7 +25,7 @@
 #include <ccMainAppInterface.h>
 
 //Qt
-#include <QMainwindow>
+#include <QMainWindow>
 
 //! Dialog for displaying the 3DSMAC/UEB disclaimer
 class TrainDisclaimerDialog : public QDialog, public Ui::TrainDisclaimerDialog
diff --git a/plugins/core/Standard/q3DMASC/q3DMASCTools.cpp b/plugins/core/Standard/q3DMASC/q3DMASCTools.cpp
index 68eb972..bfc7068 100644
--- a/plugins/core/Standard/q3DMASC/q3DMASCTools.cpp
+++ b/plugins/core/Standard/q3DMASC/q3DMASCTools.cpp
@@ -45,9 +45,6 @@
 #include <assert.h>
 #include <iostream>
 
-#if defined(_OPENMP)
-#include <omp.h>
-#endif
 using namespace masc;
 
 bool Tools::SaveClassifier(	QString filename,
@@ -1233,12 +1230,6 @@ bool Tools::PrepareFeatures(const CorePoints& corePoints, Feature::Set& features
 			CCCoreLib::NormalizedProgress nProgress(progressCb, pointCount);
 
 			QMutex mutex;
-#ifndef _DEBUG
-#if defined(_OPENMP)
-			omp_set_num_threads(std::max(1, omp_get_max_threads() - 2));
-#pragma omp parallel for
-#endif
-#endif
 			for (int i = 0; i < static_cast<int>(pointCount); ++i)
 			{
 				//spherical neighborhood extraction structure
diff --git a/plugins/core/Standard/q3DMASC/qTrain3DMASCDialog.cpp b/plugins/core/Standard/q3DMASC/qTrain3DMASCDialog.cpp
index 3959bf7..1d6fd3e 100644
--- a/plugins/core/Standard/q3DMASC/qTrain3DMASCDialog.cpp
+++ b/plugins/core/Standard/q3DMASC/qTrain3DMASCDialog.cpp
@@ -101,7 +101,7 @@ int Train3DMASCDialog::addFeature(QString name, float importance, bool isChecked
 	nameItem->setCheckState(isChecked ? Qt::Checked : Qt::Unchecked);
 	tableWidget->setItem(index, 0, nameItem);
 
-	QTableWidgetItem* importanceItem = new QTableWidgetItem(isnan(importance) ? QString() : QString::number(importance));
+	QTableWidgetItem* importanceItem = new QTableWidgetItem(std::isnan(importance) ? QString() : QString::number(importance));
 	tableWidget->setItem(index, 1, importanceItem);
 
 	return index;
@@ -175,7 +175,7 @@ void Train3DMASCDialog::setFeatureImportance(QString featureName, float importan
 		if (tableWidget->item(index, 0)->text() == featureName)
 		{
 			QTableWidgetItem* item = tableWidget->item(index, FeatureImportanceColumn);
-			item->setText(isnan(importance) ? QString() : QString::number(importance, 'f', 6));
+			item->setText(std::isnan(importance) ? QString() : QString::number(importance, 'f', 6));
 			return;
 		}
 	}